core/inc/knoting/audio_source.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <knoting/asset.h>
+#include <knoting/components.h>
+#include <knoting/types.h>
+
+#include <filesystem>
+#include <fmod.hpp>
+#include <string>
+
+namespace knot {
+class Engine;
+class AudioSubsystem;
+}  // namespace knot
+
+namespace knot::components {
+
+class AudioSource : public Asset {
+   public:
+    AudioSource(const std::filesystem::path& path = {}, bool loops = false);
+    ~AudioSource();
+
+    //=For ECS========
+    void on_awake() override;
+    void on_destroy() override;
+    //=For Asset=======
+    void generate_default_asset() override;
+    //=================
+
+    void set_path(const std::filesystem::path& path) { m_path = path; }
+
+    const std::filesystem::path& get_path() { return m_path; }
+
+    bool get_loops() { return m_loops; }
+
+    void set_loops(bool loops) { m_loops = loops; }
```

Comment:
If not special logic is happening inside encapsulations they can be removed, and the member variables made public